src/GoTrueAdminApi.ts

```diff
@@ -233,6 +233,38 @@ export default class GoTrueAdminApi {
     }
   }
 
+  /**
+   * Get user by email.
+   *
+   * @param email The user's email
+   *
+   * This function should only be called on a server. Never expose your `service_role` key in the browser.
+   */
+  async getUserByEmail(email: string): Promise<UserResponse> {
+    try {
+      const response = await _request(this.fetch, 'GET', `${this.url}/admin/users`, {
+        headers: this.headers,
+        xform: _userResponse,
+        query: {
+          filter: email,
```

Comment:
Dropped the URL Encode on email as `_request` seems to do URLEncoding [here](https://github.com/supabase/auth-js/blob/master/src/lib/fetch.ts#L156)

Passing in an encoded email seems to lead to double encoding for me locally